File: Prácticas/P5/JAVA/Civitas/src/civitas/SorpresaPagarCobrar.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author Jose Padial
 */
public class SorpresaPagarCobrar extends Sorpresa{
    private String texto;
    private int valor;
    private Tablero tablero;
    
    SorpresaPagarCobrar(Tablero tablero, int valor, String texto){
        this.tablero = tablero;
        this.valor = valor;
        this.texto = texto;
    }
    
    @Override
    void aplicarJugador(int actual, ArrayList<Jugador> todos){
        if(todos.get(actual) != null){
            informe(actual,todos);
            todos.get(actual).modificarSaldo(valor);
        }
    }

    @Override
    public String toString() {
        return "SorpresaPagarCobrar{" + "texto=" + texto + ", valor=" + valor + ", tablero=" + tablero + '}';
    }
}

File: Prácticas/P4/RUBY/Civitas/lib/sorpresa_pagar_cobrar.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative "tablero.rb"
require_relative "diario.rb"
require_relative "jugador.rb"
require_relative "mazo_sorpresas.rb"

module Civitas
  class SorpresaPagarCobrar < Sorpresa
    def initialize(tablero,valor,texto)
      @tablero = tablero
      @valor = valor
      @texto = texto
    end
    
    def aplicar_a_jugador(actual,todos)
      if(jugador_correcto(actual,todos))
        informe(actual,todos)
        todos[actual].modificar_saldo(@valor)
      end
    end
    
    def to_s
      to_s="Sorpresa pagar cobrar {#{@texto}, valor=#{@valor}}"
    end
  end
end
